src/query/plan/operator.cpp

```diff
@@ -237,6 +239,12 @@ CreateNode::CreateNodeCursor::CreateNodeCursor(const CreateNode &self, utils::Me
 bool CreateNode::CreateNodeCursor::Pull(Frame &frame, ExecutionContext &context) {
   SCOPED_PROFILE_OP("CreateNode");
 
+  if (context.auth_checker && !context.auth_checker->Accept(*context.db_accessor, self_.node_info_.labels,
+                                                            memgraph::auth::FineGrainedPermission::CREATE_DELETE)) {
+    spdlog::error("Error while creating a vertex due to not having enough permission!");
+    return false;
+  }
+
```

Comment:
do we really need to check this in every pull? not enough to do it once we know the labels?

Reply:
I talked about adding checks if user can do certain operation on nodes and edges while preparing query in interpreter.cpp with @Josipmrden and he told me that for this iteration we are going with changes only in operators.

Reply:
Yes, due to lack of time and complexity the cypher query can get, we decided to roll with only operators as we have taken a good look at them and possibly extend the functionality in the upcoming months in the interpreter if needed.

Reply:
And what about having a bool member which can be changed from `false` to `true` in the first pull and checking the privileges at the same time? I have concerns about this really plummeting the performance of create when we have privileges.